src/LDNS.xs

```diff
@@ -1375,32 +1375,42 @@ packet_edns_version(obj,...)
 # -------------------
 # Get/set EDNS data
 # 
-# Beware, this code can only take a unique U32 parameter which means it 
-# is not a full implementation of EDNS data but it is enough for our 
+# This function will set EDNS RDATA in a packet when given a parameter,
+# otherwise it will return the entire EDNS RDATA of the packet (if any).
+#
+# Beware, when setting EDNS RDATA, this code can only take a unique U32 parameter
+# which means it is not a full implementation of EDNS data but it is enough for our
 # current purpose. It can only deal with option codes with OPTION-LENGTH=0
 # (see 6.1.2 section of RFC 6891) which means OPTION-DATA is always empty.
 #
-# returns: a bytes string
+# returns: a bytes string (or undef if no EDNS data is found)
 #
 SV *
 packet_edns_data(obj,...)
     Zonemaster::LDNS::Packet obj;
     CODE:
-        ldns_rdf* opt;
+        ldns_rdf* rdf;
         if(items>=2)
         {
             SvGETMAGIC(ST(1));
-            opt = ldns_native2rdf_int32(LDNS_RDF_TYPE_INT32, (U32)SvIV(ST(1)));
-            if(opt == NULL) 
+            rdf = ldns_native2rdf_int32(LDNS_RDF_TYPE_INT32, (U32)SvIV(ST(1)));
+            if(rdf == NULL)
             {
-                croak("Failed to set OPT RDATA");
+                croak("Failed to set rdf RDATA");
             }
-            ldns_pkt_set_edns_data(obj, opt);
+            ldns_pkt_set_edns_data(obj, rdf);
+            RETVAL = newSVpvn((char*)(rdf), 4);
```

Comment:
Shouldn't this be updated to `newSVpvn((char*)(rdf->_data), 4)`?

Reply:
I'm not sure, maybe it makes more sense, but do we really care about what is returned by this function when setting edns data? Currently this will give us [_size and _type of the RDF](https://www.nlnetlabs.nl/documentation/ldns/structldns__struct__rdf.html), maybe that is more pertinent than the RDATA in this case?